auth_service/user_management/user_registry/router.py

```diff
@@ -95,7 +95,54 @@ async def post_user(
     except Exception as error:
         log.error("Could not insert user: %s", error)
         raise HTTPException(
-            status_code=400, detail="User cannot be registered."
+            status_code=500, detail="User cannot be registered."
+        ) from error
+    return user
+
+
+@router.put(
+    "/users/{id}",
+    operation_id="put_user",
+    tags=["users"],
+    summary="Update a user",
+    description="Endpoint used to update a registered user"
+    " when their name or email used by LS Login have changed."
+    " May only be performed by the users themselves."
+    " Data delivered by the OIDC provider may not be altered.",
+    responses={
+        200: {"model": User, "description": "User was successfully updated."},
+        403: {"description": "Not authorized to update user."},
+        422: {"description": "Validation error in submitted user data."},
+    },
+    status_code=200,
+)
+async def put_user(
+    user_data: UserBasicData,
+    id_: str = Path(
+        ...,
+        alias="id",
+        title="Internal ID",
+    ),
+    user_dao: UserDao = Depends(get_user_dao),
+    auth_token: AuthToken = Depends(RequireAuthToken(active=False)),
+) -> User:
+    """Update a user"""
+    # users can only update themselves,
+    # invalid users are allowed to update themselves in order to become valid again
+    if not (
+        is_internal_id(id_)
+        and id_ == auth_token.id
+        and auth_token.status in (UserStatus.ACTIVE, UserStatus.INVALID)
+    ):
+        raise HTTPException(status_code=403, detail="Not authorized to update user.")
+    try:
+        user = await user_dao.get_by_id(id_)
+        user = user.copy(update=user_data.dict())
```

Comment:
Maybe we should also check whether the data coming from LS Login is up to date with the auth_token before updating it. See line 76 - 85.

Reply:
Yes, we should do that, thanks for catching this - fixed in 4f51408e3e80f7123b608d1d7720195486ffa33f